src/esp/gfx/replay/Recorder.h

```diff
@@ -57,6 +58,15 @@ class Recorder {
    */
   void onLoadRenderAsset(const esp::assets::AssetInfo& assetInfo);
 
+  /**
+   * @brief Record deletion of all render instances in a scene graph.
+   * Because scene graphs are currently leaked when the active scene changes, we
+   * cannot rely on node deletion to issue gfx-replay deletion entries. This
+   * function allows to circumvent this issue.
+   * @param sceneGraph The scene graph to delete.
+   */
+  void onDeleteSceneGraph(const esp::scene::SceneGraph& sceneGraph);
```

Comment:
This callback name is confusing because we're not actually deleting a scene graph in the caller code (right? If we were, we wouldn't need this). I suggest a rename to `onHideSceneGraph`, `onDisableSceneGraph`, or even `onLeakSceneGraph` to capture what's really happening in Simulator.cpp. You should also revise the comment here to point people to the specific file/line in Simulator.cpp where the scene graph is leaking. Also, the description `The scene graph to delete` should be updated to something like `the scene graph being hidden/disabled/leaked`.